solver/src/main/java/DbConstraintSolverZ3InDocker.java

```diff
@@ -96,54 +104,119 @@ String solveFromFile(String fileName) {
-     */
-    private List<SqlAction> toSqlAction(String model) {
-        // Create the insert based on the first constraint
+    private Map<String, String> toSolvedConstraintsMap(String model) {
         String[] lines = model.split("\n");
-        Map<String, Gene> genes = new HashMap<>();
+        Map<String, String> solved = new HashMap<>();
 
         for (int i = 1; i < lines.length; i++) {
             String[] values = lines[i].substring(2, lines[i].length()-2).split(" ");
-            String name = values[0];
-            Integer value =  Integer.parseInt(values[1]);
-
-            Gene gene = new IntegerGene(
-                    name,
-                    value,
-                    null,
-                    null,
-                    null,
-                    false,
-                    false);
-            genes.put(name, gene);
+
+            solved.put(values[0], values[1]);
         }
+        return solved;
+    }
 
-        // TODO: Handle more than 1 table
-        String tableName = this.schemaDto.tables.get(0).name;
-        List<String> history = new LinkedList<>();
-        Set<String> columnNames = new HashSet<>(Collections.singletonList("*"));
+    private List<SqlAction> toSqlAction(Smt2Writer writer, Map<String, String> solvedConstraints) {
 
-        List<SqlAction> actions = sqlInsertBuilder.createSqlInsertionAction(tableName,
-                columnNames, history, false, false, false);
+        List<SqlAction> actions = new LinkedList<>();
+        for (TableDto table : this.schemaDto.tables) {
 
-        actions.get(0).seeTopGenes().forEach(g -> {
-            if (genes.containsKey(g.getName())) {
-                g.copyValueFrom(genes.get(g.getName()));
-            } else {
-                log.warn("Gene not found in the model: " + g.getName());
-            }
-        });
+            String tableName = table.name;
+            List<String> history = new LinkedList<>();
+            Set<String> columnNames = new HashSet<>(Collections.singletonList("*"));
+
+            List<SqlAction> newActions = sqlInsertBuilder.createSqlInsertionAction(tableName,
+                    columnNames, history, false, false, false);
+
+            newActions.forEach(action -> action.seeTopGenes().forEach(currentGene -> {
+                String tableVariableKey = writer.asTableVariableKey(table, currentGene.getName());
+                if (solvedConstraints.containsKey(tableVariableKey)) {
+
+                    if (currentGene instanceof NullableGene) {
+                        currentGene = ((NullableGene) currentGene).getGene();
+                    }
+
+                    String solvedValue = solvedConstraints.get(tableVariableKey);
+
+                    if (currentGene instanceof EnumGene) {
+                        // There is nothing to do, as the enum has the values already parsed
+                        return;
+                    }
+
+                    if (currentGene instanceof IntegerGene) {
```

Comment:
maybe not for now, but could leave a comment... aren't we missing info on constraints of the values? i mean, these genes take as input min/max info as well

Reply:
I'm not sure if I understood correctly the comment. 
Do you mean that `currentGene` already contains the info for min and max values? If that's the case, then when copying the value from the new one (`currentGene.copyValueFrom(geneWithSolvedValue)`), those values persist.

Anyway, as in `SqlBuilderFromSchemaTest` there are some cases in the max and min are not correctly parsed from the constraints, so I will address that in another PR.

Reply:
yes. based on constraints defined in schema, `currentGene` might already have info on min and max values.

Reply:
Yes, you are right. I will update only the value from the gene